shotover-proxy/tests/kafka_int_tests/test_cases.rs

```diff
@@ -539,6 +539,102 @@ pub async fn produce_consume_partitions1_kafka_node_goes_down(
     }
 }
 
+pub async fn produce_consume_partitions1_shotover_node_goes_down(
+    driver: KafkaDriver,
+    docker_compose: &DockerCompose,
+    connection_builder: &KafkaConnectionBuilder,
+    topic_name: &str,
+    shotover_node_to_kill: &str,
+) {
+    if driver.is_cpp() {
+        return;
+    }
+    {
+        let admin = connection_builder.connect_admin().await;
+        admin
+            .create_topics(&[NewTopic {
+                name: topic_name,
+                num_partitions: 1,
+                replication_factor: 3,
+            }])
+            .await;
+    }
+
+    {
+        let producer = connection_builder.connect_producer("all", 0).await;
+        // create an initial record to force kafka to create the topic if it doesnt yet exist
+        producer
+            .assert_produce(
+                Record {
+                    payload: "initial",
+                    topic_name,
+                    key: Some("Key".into()),
+                },
+                Some(0),
+            )
+            .await;
+
+        let mut consumer = connection_builder
+            .connect_consumer(
+                ConsumerConfig::consume_from_topic(topic_name.to_owned())
+                    .with_group("shotover_node_goes_down_test_group"),
+            )
+            .await;
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "initial".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(0),
+            })
+            .await;
+
+        docker_compose.kill_service(shotover_node_to_kill);
```

Comment:
shotover is not running on docker, so this is incorrect.
You will need to call `shotover.shutdown_and_then_consume_events` instead.